percy/snapshot.py

```diff
@@ -64,18 +114,24 @@ def percy_snapshot(driver, name, **kwargs):
       "For more information on usage of PercyScreenshot, "\
       "refer https://www.browserstack.com/docs/percy/integrate/functional-and-visual")
 
-
     try:
         # Inject the DOM serialization script
         driver.execute_script(fetch_percy_dom())
+        cookies = driver.get_cookies()
+        user_agent = driver.execute_script("return navigator.userAgent;")
 
         # Serialize and capture the DOM
-        dom_snapshot = driver.execute_script(f'return PercyDOM.serialize({json.dumps(kwargs)})')
+        if kwargs.get('responsive_snapshot_capture', False) or kwargs.get(
+            'responsiveSnapshotCapture', False):
+            dom_snapshot = capture_responsive_dom(driver, cookies, **kwargs)
+        else:
+            dom_snapshot = driver.execute_script(f'return PercyDOM.serialize({json.dumps(kwargs)})')
+            dom_snapshot['cookies'] = cookies
```

Comment:
is this format same as browser ? i think it wont be. The browser returns a string and this returns a dict

Reply:
I am not sure whats that ?  i am referring to cookies here which you are getting  `driver.get_cookies()`

Reply:
oh ok that, so basically in cdp we pass cookies in following format => `{ 'name': '<cookie_name>, 'value': <some_value>', ....other fields}` while document.cookies used to return string which we used to convert in above mentioned format [here](https://github.com/percy/cli/blob/dffcee04e5aefc3cc7f9f84ceaa655ad5adf9972/packages/core/src/discovery.js#L201) now `driver.get_cookies()` return hash only which is same format which needs to be passed in cdp. so now since the check typeof cookies === 'string' will not be true so cookies will be used as it is in cli which is correct format